exporter/clickhouseexporter/exporter.go

```diff
@@ -177,24 +178,57 @@ var driverName = "clickhouse" // for testing
 
 // newClickhouseClient create a clickhouse client.
 func newClickhouseClient(cfg *Config) (*sql.DB, error) {
-	// use empty database to create database
+	// create database
+	if err := createDatabase(cfg); err != nil {
+		return nil, err
+	}
+	// create table
 	db, err := sql.Open(driverName, cfg.DSN)
 	if err != nil {
 		return nil, fmt.Errorf("sql.Open:%w", err)
 	}
-	// create table
-	query := fmt.Sprintf(createLogsTableSQL, cfg.LogsTableName, "")
-	if cfg.TTLDays > 0 {
-		query = fmt.Sprintf(createLogsTableSQL,
-			cfg.LogsTableName,
-			fmt.Sprintf(`TTL toDateTime(Timestamp) + toIntervalDay(%d)`, cfg.TTLDays))
+	if _, err := db.Exec(renderCreateLogsTableSQL(cfg)); err != nil {
+		return nil, fmt.Errorf("exec create logs table sql: %w", err)
 	}
-	if _, err := db.Exec(query); err != nil {
-		return nil, fmt.Errorf("exec create table sql: %w", err)
+	if _, err := db.Exec(renderCreateTracesTableSQL(cfg)); err != nil {
+		return nil, fmt.Errorf("exec create traces table sql: %w", err)
+	}
+	if _, err := db.Exec(renderCreateTraceIDTsTableSQL(cfg)); err != nil {
+		return nil, fmt.Errorf("exec create traceIDTs table sql: %w", err)
+	}
+	if _, err := db.Exec(renderTraceIDTsMaterializedViewSQL(cfg)); err != nil {
+		return nil, fmt.Errorf("exec create traceIDTs view sql: %w", err)
```

Comment:
Should we create both logs and traces tables if only one of the pipelines is configured?

Reply:
Good point. we can add some config field about creating table for logs/traces (do_create/tablename/ ttl_days). let me create another pr about optimize config field)?

Reply:
i think i can split into initLogsDB and initTracesDB in this PR.

Reply:
Fixed in this PR, it will [create table on demond](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13442/commits/68217b3a0361b2c91aad03a75e1400d2e3df5751)

next PR i will add some config field about creating table.